flight/ODLC/point_finder.py

```diff
@@ -0,0 +1,143 @@
+"""
+Functions for finding the closest point within a boundary to a point outside a boundary
+"""
+
+from typing import List, Dict, Tuple
+from shapely.geometry import Point, Polygon
+from shapely.ops import nearest_points
+import utm
+
+
+def latlon_to_utm(coords: Dict[str, float]) -> Dict[str, float]:
+    """Converts latlon coordinates to utm coordinates and adds the data to the dictionary
+
+    Parameters
+    ----------
+    coords : Dict[str, float]
+        A dictionary containing lat long coordinates
+
+    Returns
+    -------
+    Dict[str, float]
+        An updated dictionary with additional keys and values with utm data
+    """
+
+    utm_coords = utm.from_latlon(coords["latitude"], coords["longitude"])
+    coords["utm_x"] = utm_coords[0]
+    coords["utm_y"] = utm_coords[1]
+    coords["utm_zone_number"] = utm_coords[2]
+    coords["utm_zone_letter"] = utm_coords[3]
+    return coords
+
+
+def all_latlon_to_utm(list_of_coords: List[Dict[str, float]]) -> List[Dict[str, float]]:
+    """Converts a list of dictionarys with latlon data to add utm data
+
+    Parameters
+    ----------
+    list_of_coords : List[Dict[str, float]]
+        A list of dictionaries that contain lat long data
+
+    Returns
+    -------
+    List[Dict[str, float]]
+        list[dict]: An updated list of dictionaries with added utm data
+    """
+
+    for i, _ in enumerate(list_of_coords):
+        list_of_coords[i] = latlon_to_utm(list_of_coords[i])
+    return list_of_coords
+
+
+def scale_polygon(my_polygon: Polygon, scale_factor: float = 0.1) -> Polygon:
```

Comment:
Maybe a short section after the return type, or in the function description about why 0.1 is the default for the scale factor, and if that can be or should be changed.